galsim/inclinedsersic.py

```diff
@@ -0,0 +1,216 @@
+    considering the use of only discrete n values rather than allowing it to vary continuously.  For
+    more details, see https://github.com/GalSim-developers/GalSim/issues/566.
+
+    Note that if you are building many Sersic profiles using truncation, the code will be more
+    efficient if the truncation is always the same multiple of `scale_radius`, since it caches
+    many calculations that depend on the ratio `trunc/scale_radius`.
+
+    A profile can be initialized using one (and only one) of two possible size parameters:
+    `scale_radius` or `half_light_radius`.  Exactly one of these two is required. Similarly,
+    at most one of `scale_height' and `scale_h_over_r' is required; if neither is given, the
+    default of scale_h_over_r = 0.1 will be used.
+
+    Initialization
+    --------------
+
+    @param n                The Sersic index, n.
+    @param inclination      The inclination angle, which must be a galsim.Angle instance
+    @param scale_radius     The scale radius of the exponential disk.  Typically given in arcsec.
+                            This can be compared to the 'scale_radius' parameter of the
+                            galsim.Exponential class, and in the face-on case, the same same scale
+                            radius will result in the same 2D light distribution as with that
+                            class.
+    @param scale_height     The scale height of the exponential disk.  Typically given in arcsec.
+                            [default: None]
+    @param scale_h_over_r   In lieu of the scale height, you may also specify the ratio of the
+                            scale height to the scale radius. [default: 0.1]
+    @param flux             The flux (in photons) of the profile. [default: 1]
+    @param trunc            An optional truncation radius at which the profile is made to drop to
+                            zero, in the same units as the size parameter.
+                            [default: 0, indicating no truncation]
+    @param flux_untruncated Should the provided `flux` and `half_light_radius` refer to the
+                            untruncated profile? See the documentation of the Sersic class for more
+                            details. [default: False]
+    @param gsparams         An optional GSParams argument.  See the docstring for GSParams for
+                            details. [default: None]
+
+    Methods
+    -------
+
+    In addition to the usual GSObject methods, InclinedSersic has the following access methods:
+
+        >>> n = inclined_sersic_obj.getN()
+        >>> inclination = inclined_sersic_obj.getInclination()
+        >>> r0 = inclined_sersic_obj.getScaleRadius()
+        >>> h0 = inclined_sersic_obj.getScaleHeight()
+        >>> hlr = inclined_sersic_obj.getHalfLightRadius()
+    """
+    _req_params = { "inclination" : galsim.Angle, "n" : float }
+    _opt_params = { "scale_height" : float, "scale_h_over_r" : float, "flux" : float }
```

Comment:
Shouldn't only one of `scale_height` and `scale_h_over_r` be specifiable? This currently takes in both without complaining but retains the input `scale_height`. I notice this even with the `InclinedExponential` class which has already been merged on to the master, I guess.